sprint 4 project

tigralbus opened the pull request.

Commit:
sprint 4 project commit

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,83 @@
+        collector.add_new_book('Наказание')
+        collector.add_new_book('Преступление')
+        collector.add_new_book('Наказание')
+        assert collector.books_genre == {'Преступление': '', 'Наказание': ''}
```
Нужно исправить: не проверен метод get_books_genre

tigralbus replied:
1) added tests for get_list_of_favorites_books and get_books_genre methods
2) removed 2 previous tests to get 10 tests in common (parametrized test as one)

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,83 @@
+        collector.add_book_in_favorites('Любимая_книга')
+        collector.add_book_in_favorites('Нелюбимая_книга')
+        collector.delete_book_from_favorites('Нелюбимая_книга')
+        assert collector.get_list_of_favorites_books() == ['Любимая_книга']
```
Нужно исправить: использование метода get_list_of_favorites_books в других тестах не значит что отдельный тест на него не нужен

tigralbus replied:
1) added tests for get_list_of_favorites_books and get_books_genre methods
2) removed 2 previous tests to get 10 tests in common (parametrized test as one)

Commit:
sprint 4 project:
1) added tests for get_list_of_favorites_books and get_books_genre methods
2) removed 2 previous tests to get 10 tests in common (parametrized test as one)

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,85 @@
+        collector.add_new_book('Наказание')
+        collector.add_new_book('Преступление')
+        collector.add_new_book('Наказание')
+        assert collector.books_genre == {'Преступление': '', 'Наказание': ''}
```
Нужно исправить: повторный коммент: нужен тест на проверку метода  get_books_genre(), тут его нет

tigralbus replied:
Выделенные строки относятся к тесту 3 по методу add_new_book()
Тест 4 в строках 31-35 проверяет метод get_book_genre()

AlexKitIt replied:
Коммент необходимо читать внимательно. Там указан метод get_bookS_genre() а не get_book_genre(). Недостающий тест должен быть написан. Следующая проверка будет крайней

tigralbus replied:
Спасибо за пояснение, запутало то, что названия методов  get_bookS_genre() и get_book_genre() похожи, плюс на второй тоже не было теста в первой версии проекта. Добавила 11 тест на get_bookS_genre() и обновила ридми.

tigralbus commented on tests.py:
```diff
@@ -1,24 +1,85 @@
-    # напиши свои тесты ниже
-    # чтобы тесты были независимыми в каждом из них создавай отдельный экземпляр класса BooksCollector()
\ No newline at end of file
+    # 4 test verifies that get_book_genre method return selected book genre - added 20.11
```
Проверка метода get_book_genre() добавлена здесь.

Commit:
sprint 4 project:
added test for get_bookS_genre
and readme is updated
